dotCMS/src/main/java/com/dotcms/rest/MapToContentletPopulator.java

```diff
@@ -383,53 +434,158 @@ List<Category> categories(final Contentlet contentlet, final User user, final bo
                     "Attempt to set unrecognized object [%s] as category. This field only accepts arrays like ['cat-key','cat-var','cat-inode'] or a string like 'cat-key','cat-var','cat-inode' ",
                     object));
         }
-        return categories;
+        final Map<CategoryField, Set<Category>> existingCategories = fetchExistingCategories(contentlet, user);
+        final Map<CategoryField, Set<Category>> fetchedCategories = builder.build();
+        final Map<CategoryField, Set<Category>> merged = merge(existingCategories, fetchedCategories);
+
+        if(merged.isEmpty()){
+            return Optional.empty();
+        }
+        final List<Category> collect = merged.values().stream().flatMap(Collection::stream).collect(Collectors.toList());
+
+        return Optional.of(collect);
+    }
+
+    /**
+     * here we have two maps first one comes from the db the second one comes from the request passed
+     * if a category isn't include on the request. it should remind the same. it doesnt not need to be touched or removed
+     * therefore it must be included on the result of this maps operation
+     * @param existing saved categories
+     * @param fetched fetched from the user request
+     * @return
+     */
+    final Map<CategoryField, Set<Category>> merge(final Map<CategoryField, Set<Category>> existing, final Map<CategoryField, Set<Category>> fetched){
+        if(existing.isEmpty()){
+            return fetched;
+        }
+        final Map<CategoryField, Set<Category>> builder = new HashMap<>(existing);
+        existing.forEach((categoryField, categories) -> {
+            if(fetched.containsKey(categoryField)){
+               builder.put(categoryField,fetched.get(categoryField));
+            }
+        });
+
+        return builder;
     }
 
     /**
      * given a List and a string representation of a category
      * This will try to find different criterion to get the respective category
-     * @param categories
      * @param stringValue
      * @param user
      * @param respectFrontendRoles
      */
-    private void buildUpCategories(final List<Category> categories, final String stringValue,
-            final User user, boolean respectFrontendRoles) {
-        if (UtilMethods.isSet(stringValue)) {
+    private Set<Category> getCategoriesFromStringValue(final String stringValue, final User user, boolean respectFrontendRoles) {
+        Set<Category> categories = new HashSet<>();
```

Comment:
create the set after throw the exception

Reply:
done